util/opentitan_functest_runner.sh

```diff
@@ -4,7 +4,17 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 # A shell script for executing rust-based tests for functional and e2e tests.
-# The test runner should be passed in as the first argument.
+# There are three components that make up the invocation:
+#   - test harness
+#   - arguments
+#   - test script
+# The test harness is invoked with arguments first, then the test script.
+# The test harness and test script are both passed in by environment variables.
+# Note that bazel passes user-specified test_arg arguments as arguments to this
+# script, and because the user-specified test_arg arguments come after the bazel
+# rule's arguments, they can override the ones coming from the bazel rule.
```

Comment:
Sorry, I'm a bit confused by these three lines here. Are you referring to user-specified args as arguments passed in on the command line (e.g. `bazel test //my/test/target -- user_specified_args`) or args passed in during the call to the `opentitan_functest` macro in a BUILD file?

Reply:
That is an incorrect construction for `bazel test` (perhaps you're thinking of `bazel run`?). Users pass test program args to `bazel test` with `--test_arg=user_specified_arg1 --test_arg=user_specified_arg2 ...` options.

Reply:
they don't "**override** the ones coming from the bazel rule" do they? don't they just "append" as you indicate ?

Reply:
Technically correct, but for programs that permit later options to take precedence over earlier options of the same name, it is overriding. That's why they "can" override. ;)